audb/core/load.py

```diff
@@ -22,6 +24,92 @@
 )
 
 
+def _cached_versions(
+        name: str,
+        version: str,
+        flavor: Flavor,
+        cache_root: typing.Optional[str],
+) -> typing.Sequence[typing.Tuple[LooseVersion, str, Dependencies]]:
+    r"""Find other cached versions of same flavor."""
+
+    df = cached(cache_root=cache_root)
+    df = df[df.name == name]
+
+    cached_versions = []
+    for flavor_root, row in df.iterrows():
+        if row['flavor_id'] == flavor.short_id:
+            if row['version'] == version:
+                continue
+            deps = dependencies(
+                name,
+                version=row['version'],
+                cache_root=cache_root,
+            )
+            # as it is more likely we find files
+            # in newer versions, push them to front
+            cached_versions.insert(
+                0,
+                (
+                    LooseVersion(row['version']),
+                    flavor_root,
+                    deps,
+                ),
+            )
```

Comment:
Let's say we request version `1.0.0` and we already have `0.9.0` and `1.1.0` in cache. Since the rows are incrementally  sorted by version if we don't insert to the front, we would later first check for media/table files in `0.9.0` and then in `1.1.0`. But to me it seems to make more sense to check first in `1.1.0` and then in `0.9.0` as it is more likely that we find a file files in newer versions.

Reply:
Unfortunately I cannot see this happening in the code.

```python
if row['version'] == version:
    continue
```
if we have the right version we do nothing.
If we have a wrong version (lower or higher) we add it to `cached_versions`. Are the rows returned by `audb.dependencies()` sort by the versions then?
Why do we skip versions that exactly match?

Reply:
Sorry, now I get it. The versions we are iterating are not the versions from the dependency table, but from the cache table.
And there we skip the current version and add the latest version to the top in the last loop, correct?

Reply:
Yes, exactly. We look for all versions we can possibly copy data from and we assume that it's more likely we find the files in later versions. Not sure if this makes it much faster, but once we found a file in one of the versions we can immediately copy it and do not have to check the other versions any more.